lib/client-assets.php

```diff
@@ -840,6 +840,48 @@ function gutenberg_get_available_image_sizes() {
 	return $all_sizes;
 }
 
+/**
+ * Extends block editor settings to include Gutenberg's `editor-styles.css` as
+ * taking precedent those styles shipped with core.
+ *
+ * @param array $settings Default editor settings.
+ *
+ * @return array Filtered editor settings.
+ */
+function gutenberg_extend_block_editor_styles( $settings ) {
+	if ( empty( $settings['styles'] ) ) {
+		$settings['styles'] = array();
+	} else {
+		/*
+		 * By handling this filter at an earlier-than-default priority and with
+		 * an understanding that plugins should concatenate (not unshift) their
+		 * own custom styles, it's assumed that the first entry of the styles
+		 * setting should be the default (core) stylesheet.
+		 */
+		array_shift( $settings['styles'] );
```

Comment:
I wonder if it's simpler to just loop over the styles are replace the core editor styles if found (checking the path or something).

Reply:
> replace the core editor styles if found (checking the path or something).

The problem is that the array contains only the CSS string, no file paths.

It looks something like:

```
$settings = array(
	'styles' => array(
		array( 'css' => 'body { /* ... */ }' )
	)
);
```

So we've got nothing to work from to know whether it's the core style, unless we explicitly (re-)read the `ABSPATH . WPINC . '/css/dist/editor/editor-styles.css'` file from disk and compare it with the array style contents.

Reply:
Let's just do that.